src/Recipe.js

```diff
@@ -21,22 +21,46 @@ class Recipe {
     return ingredientPrice;
   }
 
-  getIngredientNameList(ingredients) {
-    let ingredientIds = this.ingredients.map(ingredient => {
+  getIngredientUnits(ingredients) {
+    return ingredients.map(ingredient => {
+      return {
+        [ingredient.id]: ingredient.quantity.unit
+      }
+    })
+  }
+
+  getIngredientIds(ingredients) {
+    return ingredientIds = ingredients.map(ingredient => {
       return ingredient.id;
     })
-    let ingredientObjects = ingredientIds.map(id => {
-      return ingredientsData.find(ingredient => {
-        return id === ingredient.id;
-      });
-    });
+  }
+
+  getIngredientObjects(ingredientPiece) {
```

Comment:
I don't see a difference between the if and the else statements, I think this could be cleaned up if the code works.